File: 프로그래머스/0/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = {};
        
        for(int [] ar : queries) {
            int x = ar[0];
            int y = ar[1];
            int z = ar[2];
            
            for(int i=x; i<=y; i++) {
                if(i % z == 0) arr[i] += 1;
                
            }
        }
        
        return arr;
    }
}

File: 프로그래머스/0/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] queries) {
        int[] answer = (int[])arr.Clone();
        int leng = queries.GetLength(0);
        
        for(int i = 0; i < leng; i++)
        {
            for(int j = queries[i, 0]; j <= queries[i,1]; j++)
            {
                if(j % queries[i, 2] == 0)
                    answer[j]++;
            }
        }
        
        return answer;
    }
}